includes/utilities.inc

```diff
@@ -104,12 +104,29 @@ function islandora_rest_accessible($endpoint, $request_method, $resource = FALSE
   if ($endpoint == 'solr') {
     return islandora_rest_solr_access();
   }
+
+  // Retrieve both islandora core and islandora rest based permissions.
   $perm = islandora_rest_resource_access_permission($endpoint, $request_method);
+  $access = array();
   if (is_subclass_of($resource, 'AbstractObject')) {
-    return islandora_object_access($perm, $resource);
+    // Call core permission check, to ensure XACML permissions are
+    // Respected.
+    array_push($access, islandora_object_access($perm['core'], $resource));
+    array_push($access, islandora_object_access($perm['rest'], $resource));
+
+    // Deny access if either permission returns no access.
+    return in_array(FALSE, $access) ? FALSE : TRUE;
   }
   elseif (is_subclass_of($resource, 'AbstractDatastream')) {
-    return islandora_datastream_access($perm, $resource);
+    // Only for use with islandora tokened object access permission.
+    if (is_callable($perm['core'])) {
+      array_push($access, islandora_object_datastream_tokened_access_callback(ISLANDORA_VIEW_OBJECTS, $resource->parent->id, $resource));
```

Comment:
Kind of _very_ hackish (and doesn't make use of the callable that is `$perm['core']`). Also, I think don't think this is really appropriate... We want to check if we're allowed to _generate_ tokens, not accept one in the URL. Can probably just map `ISLANDORA_REST_DATASTREAM_TOKEN_GET_PERM` to `ISLANDORA_VIEW_OBJECTS`.